src/test/java/reposense/git/GitCommitEditerTest.java

```diff
@@ -0,0 +1,86 @@
+package reposense.git;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import reposense.model.Author;
+import reposense.parser.InvalidLocationException;
+import reposense.system.CommandRunner;
+import reposense.template.GitTestTemplate;
+import reposense.util.TestUtil;
+
+public class GitCommitEditerTest extends GitTestTemplate {
+    private static final int NUMBER_FAKE_AUTHOR_COMMIT = 4;
+    private static final int NUMBER_EUGENE_AUTHOR_COMMIT = 1;
+
+    @Before
+    @Override
+    public void before() throws InvalidLocationException {
+        super.before();
+        // create new temp branch so as to not affect the master branch for other tests
+        CommandRunner.checkoutNewBranch(config.getRepoRoot(), TEMP_BRANCH);
```

Comment:
perhaps we should do this in the `GitTestTemplate` itself?

Reply:
Sounds good, ensure all tests don't affect the original "master" in anyway